src/utils/addresses.ts

```diff
@@ -16,10 +16,54 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
+export type AddressInfo = {
+  index: number
+  isMain: boolean
+  label?: string
+  color?: string
+}
+
 export const checkAddressValidity = (address: string) => {
   const match = address.match(/^[1-9A-HJ-NP-Za-km-z]+$/)
 
   if (match === null) return false
 
   return match[0] === address && address
 }
+
+export const loadStoredAddressesInfoOfAccount = (username: string) => {
+  const storedAddressIndexes = localStorage.getItem(`${username}-address-info`)
+
+  if (storedAddressIndexes === null) return []
+
+  return JSON.parse(storedAddressIndexes)
+}
+
+export const storeAddressInfoOfAccount = (
+  addressIndex: number,
+  username: string,
+  label?: string,
+  color?: string,
+  isMain = false
+) => {
+  const addressesInfo = loadStoredAddressesInfoOfAccount(username)
+  const existingAddressInfo = addressesInfo.find((info: AddressInfo) => info.index === addressIndex)
+
+  if (!existingAddressInfo) {
+    addressesInfo.push({
+      index: addressIndex,
+      label,
+      color,
+      isMain
+    })
+  } else if (
+    existingAddressInfo.label !== label ||
+    existingAddressInfo.color !== color ||
+    existingAddressInfo.isMain !== isMain
+  ) {
+    existingAddressInfo.label = label
+    existingAddressInfo.color = color
+    existingAddressInfo.isMain = isMain
+  }
```

Comment:
The "else if" condition can be simplified greatly with `Object.assign(target, source)` to automatically merge changes, as long as the object as a depth of 1. 

Something like:
```
else {
 newAddressInfo = Object.assign({label, color, isMain}, existingAddressInfo)
}
```

Reply:
Done!